engine/pkg/externalresource/manager/gc_runner.go

```diff
@@ -136,7 +138,13 @@ func (r *DefaultGCRunner) gcOnce(
 	}
 
 	if err := handler(ctx, res); err != nil {
-		return err
+		st := status.Convert(err)
+		if st.Code() != codes.NotFound {
+			return err
+		}
```

Comment:
We will have different resource handler in the future. 
May be we can define some common error code for gc handler, instead of using the grpc error here?
For example, when we use S3, will we encounter another error except the `NotFound`?